example/view/main.go

```diff
@@ -0,0 +1,116 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:build go1.18
+// +build go1.18
+
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+
+	"github.com/prometheus/client_golang/prometheus"
+	"github.com/prometheus/client_golang/prometheus/promhttp"
+
+	"go.opentelemetry.io/otel/attribute"
+	otelprom "go.opentelemetry.io/otel/exporters/prometheus"
+	"go.opentelemetry.io/otel/metric/instrument"
+	"go.opentelemetry.io/otel/sdk/instrumentation"
+	"go.opentelemetry.io/otel/sdk/metric"
+	"go.opentelemetry.io/otel/sdk/metric/aggregation"
+	"go.opentelemetry.io/otel/sdk/metric/view"
+)
+
+const meterName = "github.com/open-telemetry/opentelemetry-go/example/view"
+
+func main() {
+	ctx := context.Background()
+
+	// The exporter embeds a default OpenTelemetry Reader, allowing it to be used in WithReader
+	exporter := otelprom.New()
+
+	// View to customize histogram
+	customBucketsView, err := view.New(
+		// Match* to match instruments
+		view.MatchInstrumentName("custom_histogram"),
+		view.MatchInstrumentKind(view.SyncHistogram),
+		view.MatchInstrumentationScope(instrumentation.Scope{Name: meterName}),
+
+		// With* to modify instruments
+		view.WithSetAggregation(aggregation.ExplicitBucketHistogram{
+			Boundaries: []float64{64, 128, 256, 512, 1024, 2048, 4096},
+		}),
+		view.WithRename("bar"),
+	)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	// Default view to keep all instruments
+	defaultView, err := view.New(view.MatchInstrumentName("*"))
+	if err != nil {
+		log.Fatal(err)
+	}
```

Comment:
Note, this shouldn't be needed. Tracking resolution with https://github.com/open-telemetry/opentelemetry-go/issues/3224